test/gtest/ucp/ucp_test.h

```diff
@@ -212,6 +213,48 @@ class ucp_test : public ucp_test_base,
     volatile int m_err_handler_count;
     static const ucp_datatype_t DATATYPE;
     static const ucp_datatype_t DATATYPE_IOV;
+
+protected:
+    class mapped_buffer {
+    public:
+        mapped_buffer(size_t size, const entity& entity, int flags = 0,
+                      uint64_t seed = 0,
+                      ucs_memory_type_t mem_type = UCS_MEMORY_TYPE_HOST);
+        virtual ~mapped_buffer();
+
+        void *ptr() const;
+        uintptr_t addr() const;
+        size_t length() const;
+        ucs_memory_type_t mem_type() const;
+
+        void pattern_fill(uint64_t seed);
+        void pattern_check(uint64_t seed);
+        void pattern_fill();
+        void pattern_check();
+
+        ucp_rkey_h rkey(const entity& entity) const;
+        ucp_mem_h memh() const;
+
+        class ucp_rkey {
```

Comment:
use ucs::handle<ucp_rkey_h> instead of defining this class

Reply:
fixed